sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -83,52 +103,23 @@ export class CryptographyClient {
     plaintext: Uint8Array,
     options: EncryptOptions = {}
   ): Promise<EncryptResult> {
+    const localCryptographyClient = await this.getLocalCryptographyClient();
     const requestOptions = operationOptionsToRequestOptionsBase(options);
     const span = this.createSpan("encrypt", requestOptions);
 
-    if (isNode) {
-      await this.fetchFullKeyIfPossible();
-
-      if (typeof this.key !== "string") {
-        switch (algorithm) {
-          case "RSA1_5": {
-            if (this.key.kty !== "RSA") {
-              span.end();
-              throw new Error("Key type does not match algorithm");
-            }
-
-            if (this.key.keyOps && !this.key.keyOps.includes("encrypt")) {
-              span.end();
-              throw new Error("Key does not support the encrypt operation");
-            }
-
-            const keyPEM = convertJWKtoPEM(this.key);
-
-            const padded: any = { key: keyPEM, padding: constants.RSA_PKCS1_PADDING };
-            const encrypted = publicEncrypt(padded, Buffer.from(plaintext));
-            return { result: encrypted, algorithm, keyID: this.key.kid };
-          }
-          case "RSA-OAEP": {
-            if (this.key.kty !== "RSA") {
-              span.end();
-              throw new Error("Key type does not match algorithm");
-            }
-
-            if (this.key.keyOps && !this.key.keyOps.includes("encrypt")) {
-              span.end();
-              throw new Error("Key does not support the encrypt operation");
-            }
-
-            const keyPEM = convertJWKtoPEM(this.key);
-
-            const encrypted = publicEncrypt(keyPEM, Buffer.from(plaintext));
-            return { result: encrypted, algorithm, keyID: this.key.kid };
-          }
+    if (localCryptographyClient && isLocallySupported(algorithm)) {
```

Comment:
This also needs to check if the keyPermissions also supports the operation (if non-empty) and if the key is validate given its `NotBefore` and `ExpiresOn` values, if present.

Reply:
ohhh thank you!

Reply:
@heaths I'm confused,

Are the permissions on the KeyVaultKey.Key (JsonWebKey representation) as the KeyOps? Or are we missing something?

Here's the approach that I'll be taking for now:

```ts
  private async checkPermissions(operation: KeyOperation): Promise<void> {
    await this.getLocalCryptographyClient();

    if (typeof this.key !== "string" && this.key.keyOps && !this.key.keyOps.includes(operation)) {
      throw new Error(`Operation ${operation} is not supported on key ${this.getKeyID()}`);
    }
  }

  private async checkKeyValidity(): Promise<void> {
    await this.getLocalCryptographyClient();
    const attributes = this.keyBundle?.attributes || {};
    const { notBefore, expires } = attributes;
    const now = new Date();

    if (notBefore && now < notBefore) {
      throw new Error(`Key ${this.getKeyID()} can't be used before ${notBefore}`);
    }

    if (expires && now < expires) {
      throw new Error(`Key ${this.getKeyID()} expired at ${expires}`);      
    }
```

I will have this pushed to this PR, your feedback is welcome either in this comment thread or in the actual changes 🙌

Reply:
Close, but you might want to check if `now > expires`.